app/models/ext_management_system.rb

```diff
@@ -442,6 +444,9 @@ def self.schedule_destroy_queue(id, deliver_on = nil)
       :method_name => "orchestrate_destroy",
       :deliver_on  => deliver_on,
     )
+    find(id).child_managers.each do |child_manager|
```

Comment:
How about doing this in the instance method so you can skip the `find(id)` and just have access to `child_managers`?

Reply:
You would think `def destroy_queue` is called, but alas [not](https://github.com/ManageIQ/manageiq-ui-classic/blob/65c13016f891f1232cffa43309d552a810cc6ccc/app/controllers/ems_common.rb#L547) The UI calls it on the model. And as the instance method also forwards it to the class method, it's safer to do it here.